subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -953,23 +915,77 @@ static void pe_chunk_received_run(void *obj)
 	}
 }
 
-/**
- * @brief Suspend State
+/*
+ * @brief Super State for any message that requires
+ *	  Sender Response Timer functionality
  */
-static void pe_suspend_entry(void *obj)
+static void pe_sender_response_run(void *obj)
 {
-	LOG_INF("PE_SUSPEND");
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	enum usbc_pe_state current_state = pe_get_state(dev);
+
+	/* Start the Sender Response Timer after the message is sent */
+	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		/* Start Sender Response Timer */
+		usbc_timer_start(&pe->pd_t_sender_response);
+	}
+
+	/* Check if the Sender Response Timer has expired */
+	if (usbc_timer_expired(&pe->pd_t_sender_response)) {
+		/* Inform the DPM of the sender response timeout */
+		policy_notify(dev, SENDER_RESPONSE_TIMEOUT);
+
+		/*
+		 * Handle Sender Response Timeouts
+		 */
+		switch (current_state) {
+		/* Sink states */
+		case PE_SNK_SELECT_CAPABILITY:
+			pe_set_state(dev, PE_SNK_HARD_RESET);
+			break;
+		case PE_SNK_GET_SOURCE_CAP:
+			pe_set_state(dev, PE_SNK_READY);
+			break;
+
+		/*
+		 * Common states:
+		 * Could transition to a Sink or Source states,
+		 * depending on the current Data Role
+		 */
+		case PE_SOFT_RESET:
```

Comment:
This condition is handled in the base state, static void pe_sender_response_run(void *obj)
```
else if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_MSG_RECEIVED)) {
		/*
		 * The Policy Engine Shall transition to the PE_SNK_Wait_for_Capabilities
		 * state when:
		 *      1: An Accept Message has been received on SOP
		 */
		header = prl_rx->emsg.header;

		if (received_control_message(dev, header, PD_CTRL_ACCEPT)) {
			pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
		}
```

Reply:
I think this comment train is talking about two different states.  8.3.3.4.2.1 PE_SNK_Send_Soft_Reset State initiates the SenderResponseTimer after sending the soft reset, whereas the very similarly named 8.3.3.4.2.2 PE_SNK_Soft_Reset State does not need a SenderResponseTimer since it's just sending the response of Accept.

Reply:
Correct, `PE_SNK_Soft_Reset` doesn't need SenderResponseTimer, so the `PE_SEND_SOFT_RESET` state should leave it's parent state set to NULL.

Reply:
Sorry for my confusion. I've made the change.